forui/lib/src/widgets/separator/separator.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:forui/forui.dart';
+
+/// A separator visually separates content.
+final class FSeparator extends StatelessWidget {
+
+  /// The style.
+  final FSeparatorStyle? style;
+
+  /// Whether this separator should be horizontal or vertical. Defaults to horizontal (false).
+  final bool vertical;
+
+  /// Creates a [FSeparator].
+  const FSeparator({this.style, this.vertical = false, super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final style = this.style ?? (vertical ? context.theme.separatorStyles.vertical : context.theme.separatorStyles.horizontal);
+    return Padding(
+      padding: style.padding,
+      child: vertical ?
+        SizedBox(
+          width: style.thickness,
+          child: ColoredBox(
+            color: style.color,
+          ),
+        ) :
+        SizedBox(
+          height: style.thickness,
+          child: ColoredBox(
+            color: style.color,
+          ),
+        ),
+    );
+  }
+
+}
+
+/// [FSeparator]'s style.
+final class FSeparatorStyle with Diagnosticable {
+
+  /// The default padding for horizontal and vertical separators.
+  static const defaultPadding = (
+    horizontal: EdgeInsets.symmetric(vertical: 20),
+    vertical: EdgeInsets.symmetric(horizontal: 20),
+  );
+
+  /// The color of the separating line. Defaults to [FColorScheme.secondary].
+  final Color color;
+
+  /// The padding surrounding the separating line.
+  final EdgeInsetsGeometry padding;
+
+  /// The thickness of the separating line. Defaults to 1.
+  ///
+  /// ## Contract:
+  /// Throws an [AssertionError] if:
+  /// * `thickness` <= 0.0
+  /// * `thickness` is Nan
+  final double thickness;
```

Comment:
Do you think it's a good idea to add this into the global style? (in addition to `borderRadius`) Could be useful for `Card` and `Button` too.

Also, it might be a good idea to follow the CSS naming convention for this field:
https://developer.mozilla.org/en-US/docs/Web/CSS/border-width
https://tailwindcss.com/docs/border-width